### PR TITLE
[parachain-template] benchmarks into `mod benchmarks`

### PR DESCRIPTION
**Pull request by asiniscalchi**
This PR introduces a dedicated module for benchmarks within the parachain runtime. By segregating benchmarks into their own module, we achieve a cleaner project structure and improved readability, facilitating easier maintenance and updates.

### Key Changes:
- **New Benchmarks Module**: A new file `benchmarks.rs` is added, encapsulating the benchmarking code for various pallets. 
- **Refactoring `lib.rs`**: The main runtime library file (`lib.rs`) has been updated to reflect the extraction of benchmark definitions. By moving these definitions to `benchmarks.rs`, we reduce clutter in `lib.rs`, streamlining the runtime's core logic and configuration.

### Benefits of This Refactoring:
- **Focused Benchmarking**: Developers can now easily locate and modify benchmarks without navigating through the core runtime logic, enabling targeted performance improvements.
- **Cleaner Codebase**: Segregating benchmarks from the main runtime logic helps maintain a clean, well-organized codebase, simplifying navigation and maintenance.
- **Scalability**: As the parachain evolves, adding or updating benchmarks becomes more straightforward, supporting scalability and adaptability of the runtime.

### Summary of Changes:
- Created `benchmarks.rs` to house the benchmarking suite.
- Streamlined `lib.rs` by removing the inlined benchmark definitions and linking to the new benchmarks module.

### CODE REVIEW

**Review by ggwpez — COMMENTED** [review]
I dont see the value in this change.

Please consider that reviewers have very limited. This time can either be spent on reviewing important changes, or it can be wasted with unimportant changes. Either way; it is a **limited resource**, that must be deployed wisely for a best overall outcome.

File: templates/parachain/runtime/src/benchmarks.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+frame_benchmarking::define_benchmarks!(
```
Comment (ggwpez) [inline templates/parachain/runtime/src/benchmarks.rs:1]: License header missing

**Review by ggwpez — APPROVED** [review]
Okay approving as it could help new teams.

File: templates/parachain/runtime/src/benchmarks.rs (hunk 1 above)
Comment (asiniscalchi) [reply templates/parachain/runtime/src/benchmarks.rs:26]: done

**Comment by asiniscalchi** [comment]
@bkchr ping :)

File: templates/parachain/runtime/src/benchmarks.rs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+// OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+//
+// For more information, please refer to <http://unlicense.org>
```
[the remark this answers is not in the document]
Comment (asiniscalchi) [reply templates/parachain/runtime/src/benchmarks.rs:24]: https://github.com/paritytech/polkadot-sdk/blob/master/templates/parachain/LICENSE

maybe a user story ( issue ) will force a resolution

**Comment by asiniscalchi** [comment]
in a way you pushed me to publish: 
https://medium.com/@asiniscalchi/refactoring-for-entropy-how-thermodynamics-illuminates-software-development-d86089add7b6

**Comment by asiniscalchi** [comment]
@kianenigma maybe this will make sense as part of https://github.com/paritytech/polkadot-sdk/issues/3819

**Review by kianenigma — APPROVED** [review]

**Comment by asiniscalchi** [comment]
@kianenigma I think T0 is missing

**Comment by asiniscalchi** [comment]
@kianenigma ... last merge step

**Pull request merged**